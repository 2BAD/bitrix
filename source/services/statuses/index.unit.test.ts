import Statuses from '.'

const mockCall = jest.fn(() => Promise.resolve()) as any
const mockList = jest.fn(() => Promise.resolve()) as any
const statuses = Statuses({ call: mockCall })

const ID = '77'

describe('Statuses', () => {
  beforeEach(() => {
    mockCall.mockClear()
    mockList.mockClear()
  })

  it('should expose API methods', () => {
    expect(statuses).toMatchSnapshot()
  })

  describe('`create`', () => {
    it('should invoke `call`', async () => {
      await statuses.create({
        NAME: 'Test name'
      })
      expect(mockCall.mock.calls).toMatchSnapshot()
    })
  })

  describe('`get`', () => {
    it('should invoke `call`', async () => {
      await statuses.get(ID)
      expect(mockCall.mock.calls).toMatchSnapshot()
    })
  })

  describe('`list`', () => {
    it('should invoke `call`', async () => {
      await statuses.list({
        order: { NAME: 'ASC' },
        filter: { ENTITY_ID: 'STATUS' }
      })
      expect(mockCall.mock.calls).toMatchSnapshot()
    })

    it('should work without optional params', async () => {
      await statuses.list()
      expect(mockCall.mock.calls).toMatchSnapshot()
    })
  })

  describe('`update`', () => {
    it('should invoke `call`', async () => {
      await statuses.update(ID, {
        NAME: 'Test name'
      })
      expect(mockCall.mock.calls).toMatchSnapshot()
    })

    it('should work without optional params', async () => {
      await statuses.update(ID, {
        NAME: 'Test name'
      })
      expect(mockCall.mock.calls).toMatchSnapshot()
    })
  })
})
