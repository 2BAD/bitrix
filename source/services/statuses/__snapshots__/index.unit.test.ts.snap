// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Statuses \`create\` should invoke \`call\` 1`] = `
Array [
  Array [
    "crm.status.add",
    Object {
      "fields": Object {
        "NAME": "Test name",
      },
    },
  ],
]
`;

exports[`Statuses \`get\` should invoke \`call\` 1`] = `
Array [
  Array [
    "crm.status.get",
    Object {
      "id": "77",
    },
  ],
]
`;

exports[`Statuses \`list\` should invoke \`list\` 1`] = `Array []`;

exports[`Statuses \`list\` should work without optional params 1`] = `Array []`;

exports[`Statuses \`update\` should invoke \`call\` 1`] = `
Array [
  Array [
    "crm.status.update",
    Object {
      "fields": Object {
        "NAME": "Test name",
      },
      "id": "77",
    },
  ],
]
`;

exports[`Statuses \`update\` should work without optional params 1`] = `
Array [
  Array [
    "crm.status.update",
    Object {
      "fields": Object {
        "NAME": "Test name",
      },
      "id": "77",
    },
  ],
]
`;

exports[`Statuses should expose API methods 1`] = `
Object {
  "create": [Function],
  "get": [Function],
  "list": [Function],
  "update": [Function],
}
`;
